StateMachine.cpp

```diff
@@ -53,12 +53,17 @@ void StateMachine::ExternalEvent(BYTE newState, const EventData* pData)
 //----------------------------------------------------------------------------
 void StateMachine::InternalEvent(BYTE newState, const EventData* pData)
 {
-	if (pData == NULL)
+	// if (pData == NULL)
+	// 	pData = new NoEventData();
+#ifndef EXTERNAL_EVENT_NO_HEAP_DATA
+  if (pData == NULL) {
 		pData = new NoEventData();
```

Comment:
In line 127, there is a delete call we should also remove. When I compile the code for unit test with `EXTERNAL_EVENT_NO_HEAP_DATA`, the unit test failed due to the `delete` call. I don't know why it tries to delete something when we specified `EXTERNAL_EVENT_NO_HEAP_DATA`. I was able to get my unit test working by removing the delete call. Please help to check it.

Reply:
The library originally always used heap for _internal_ events; our updates are changing that behavior.
Can you add the code suggestion here?

Reply:
At line 127, i justg remove the delete part. 
```suggestion
if (pDataTemp)
{
	//if (!externalEvent)
	//	delete pDataTemp;
	pDataTemp = NULL;
}
```